### PR TITLE
Webcon get monitor initial positions (on load and kicker value change)

### LINKED ISSUE #2837: SRW beamline Initial Intensity 400 Bad Request
Opened the "NSLS-II ESM beamline" example and selected the Beamline page the first report tried to run and then showed a "400 Bad Request" error. Revisiting the report and it ran OK.

```
sirepo/job_api.py:202:_request api=api_runSimulation runDir=/srv/sirepo/db/user/Prj7BeAZ/srw/J4v7heMT/initialIntensityReport
"POST /run-simulation HTTP/1.1" 200 91 "https://alpha.sirepo.com/srw" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10.15; rv:79.0) Gecko/20100101 Firefox/79.0"'
"POST /save-simulation HTTP/1.1" 200 11148 "https://alpha.sirepo.com/srw" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10.15; rv:79.0) Gecko/20100101 Firefox/79.0"'
sirepo/http_request.py:40:parse_json content-type is not application/json: mimetype=
sirepo/uri_router.py:217:_dispatch exception=400 Bad Request: The browser (or proxy) sent a request that this server could not understand. path=run-status stack=Traceback (most recent call last):
/sirepo/uri_router.py", line 215, in _dispatch
return call_api(route.func, kwargs)
/sirepo/uri_router.py", line 88, in call_api
r = sirepo.http_reply.gen_exception(e)
/sirepo/http_reply.py", line 59, in gen_exception
return _gen_exception_werkzeug(exc)
/sirepo/http_reply.py", line 377, in _gen_exception_werkzeug
raise exc
/sirepo/uri_router.py", line 79, in call_api
r = flask.make_response(f(**kwargs) if kwargs else f())
/sirepo/job_api.py", line 148, in api_runStatus
return _request()
/sirepo/job_api.py", line 197, in _request
c = k.pkdel('_request_content') if '_request_content' in k else _request_content(k)
/sirepo/job_api.py", line 223, in _request_content
check_sim_exists=True,
/sirepo/http_request.py", line 84, in parse_post
r = parse_json()
/sirepo/http_request.py", line 40, in parse_json
req.mimetype,
/sirepo/util.py", line 194, in raise_bad_request
_raise('BadRequest', *args, **kwargs)
/sirepo/util.py", line 238, in _raise
raise getattr(werkzeug.exceptions, exc)()
werkzeug.exceptions.BadRequest: 400 Bad Request: The browser (or proxy) sent a request that this server could not understand.
Exception was printed at:
werkzeug.exceptions.BadRequest: 400 Bad Request: The browser (or proxy) sent a request that this server could not understand.
/flask/app.py", line 2464, in __call__
return self.wsgi_app(environ, start_response)
/flask/app.py", line 2447, in wsgi_app
response = self.full_dispatch_request()
/flask/app.py", line 1950, in full_dispatch_request
rv = self.dispatch_request()
/flask/app.py", line 1936, in dispatch_request
return self.view_functions[rule.endpoint](**req.view_args)
"GET /run-status HTTP/1.1" 400 281 "https://alpha.sirepo.com/srw" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10.15; rv:79.0) Gecko/20100101 Firefox/79.0"'
```

### LINKED ISSUE #3596: webcon: allow updating/running sim outside of optimizer
When the optimization isn't running, it should be possible to update a kicker value which should run the mad-x simulation one time and update the monitors. One way to do this would be to respond to any beamline element updates and immediately kick-off the simulation.

### LINKED ISSUE #3597: webcon: run simulation once initially after mad-x lattice selection
After the mad-x lattice has been selected, immediately run the simulation one time to get the monitor positions. Related to #3596

### PR DESCRIPTION
**Pull request by e-carlin**
- Fix #3597: run simulation once after mad-x lattice selection
- Fix #3596: Run simulation once when kicker values are updated 
- Fix #2837: resolve 400 response for some runSimulation requests

### CODE REVIEW

**Comment by moellep** [comment]
The kicker elements aren't accepting input - the Current field is disabled. It should be disabled only when an optimization is in process, and available for editing otherwise. Changing a kicker Current should do the run-once simulation. It would also be nice if the UI showed a hint that a run-once simulation was running, possibly animating the border around the kicker fields similar to when an optimization is in progress.

**Comment by e-carlin** [comment]
@moellep I've resolved your comments. The Kicker current elements were always disabled so I've updated the logic to be like the quads (only when optimization is running). I've also made it so the animated border is visible when the initial positions are being calculated.

File: sirepo/package_data/static/js/controls.js [hunk 1]
```diff
@@ -219,6 +244,33 @@ SIREPO.app.controller('ControlsController', function(appState, controlsService,
         return res;
     }
 
+    function getInitialMonitorPositions() {
+        const o = new MutationObserver(function (mutations) {
```
Comment (e-carlin) [inline sirepo/package_data/static/js/controls.js:248]: @moellep I have to imagine there is a better way to do this. What I want to do is what until the readout-table is rendered in the DOM before trying to apply the classes which would be a no-op. We have things like panelState.waitForUI but nothing besides this seemed to work outside of a single directive. I also tried using rootScope.broadcast and a scope.on here but the broadcast happens multiple times and I de-registering the listener seemed to not always work.

Comment (moellep) [reply sirepo/package_data/static/js/controls.js]: I changed it to a boolean: controlsService.isReadoutTableActive and then refer to that value whenever the readout table is converted to html.

**Pull request merged**